### PR TITLE
fix(prometheus): Disable auth for prometheus endpoint

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

**Review by thalesmg — COMMENTED** [review]
Maybe this should be configurable?  Otherwise, might leak information from that endpoint?

**Comment by ieQu1** [comment]
> Maybe this should be configurable? Otherwise, might leak information from that endpoint?

EMQX doesn't allow to finetune user capabilities at this moment, so there are two alternatives:

1) Expose metrics unprotected
2) Leak admin password to prometheus

I think 1 is lesser evil.

**Review by thalesmg — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

File: CHANGES-4.3.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ File format:
   should return false if match it to a topic.
 - Added a test to prevent a last will testament message to be
   published when a client is denied connection. [#8894](https://github.com/emqx/emqx/pull/8894)
+- Disable authorization for `api/v4/emqx_prometheus` endpoint. [8955](https://github.com/emqx/emqx/pull/8955)
```
Comment (zmstone) [inline CHANGES-4.3.md:29]: this change is not going into 4.3.20,
please add a 4.3.21

File: lib-ce/emqx_dashboard/src/emqx_dashboard.erl [hunk 2]
```diff
@@ -109,6 +109,8 @@ http_handlers() ->
 is_authorized(Req) ->
     is_authorized(binary_to_list(cowboy_req:path(Req)), Req).
 
+is_authorized("/api/v4/emqx_prometheus", _Req) ->
```
Comment (zmstone) [inline lib-ce/emqx_dashboard/src/emqx_dashboard.erl:112]: Should check if method is GET?

**Comment by qammmm** [comment]
The test passed with no problem

**Pull request merged**